### ISSUE TITLE
Dropout / Patience (EarlyStopping) feature consideration

### ISSUE THREAD

**Issue by cameronjacobson** [keras-team/keras#14554]
When I saw the most recent epoch output from a model that had finished training this evening it struck me it may be worth entertaining the idea that Dropout should impact Patience (or perhaps only on an "opt-in" basis).  I happened to see the model training in the background on one of my screens when it finished, and it appeared obvious, based on the log I could see on the screen, that Dropout had, JUST a few epochs prior, dropped 1 or more nodes.  It's somewhat of an assumption not knowing the inner workings Keras / Tensorflow in great detail, but I saw the LOSS jump up a bit (well more than a typical epoch at that stage of training) which made me think a node probably dropped out to make that happen.  But it occurred to me that I really didn't want the patience parameter to kick in at that moment.  I would've preferred to allow that model to continue training until there's no more dropout within what I specified for the patience parameter.  I am thinking it may be more intuitive if Dropout (if it happens to drop one or more nodes at any point during training) SHOULD be able to reset the "patience" countdown on the fly (just as the "tallied metric" does when it improves by at least min_delta within the same period of time).  The most recent dropout had no ability to impact the training because previous training had achieved a higher score (in this case) ~150 epochs prior.

This is assuming that once a node is dropped it stays dropped (which from my understanding is the case).  Otherwise I imagine you could get stuck in a back and forth where patience keeps resetting because of one or more nodes keep toggling on / off over a period of time less than the number of epochs specified in the patience setting.

**Comment by cameronjacobson** [comment]
A while back I had seen a talk online where the speaker used the term Dropout.  They used it to describe a layer which "dropped" nodes if the activation function didn't meet a minimum threshold.  I realize now the terminology may have been used inaccurately or maybe it has multiple meanings in ML.  I have since read a more detailed description of Keras dropout layers which helped me realize this ticket isn't relevant.
